### PR TITLE
Move `undefined` test into store.handle

### LINKED ISSUE #37: getInitialState: () => null causes failures
Steps to reproduce:

1) Return `null` from a Store's `getInitialState`.
2) Dispatch an action that the store doesn't handle.

Expected result:

The store ignores the action.

Actual result:

`Store handler must return a value, did you forget a return statement`

### PR DESCRIPTION
**Pull request by appsforartists**
Fixes #37

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -45,13 +45,17 @@ class Store {
...
   handle(state, type, payload) {
+    var result = state
     var handler = this.__handlers.get(type)
 
     if (typeof handler === 'function') {
-      return handler.call(this, state, payload, type)
+      result = handler.call(this, state, payload, type)
+
+      if (result === undefined)
+        throw new Error("Store handler must return a value, did you forget a return statement")
```
Comment (loganlinn) [inline src/store.js]: minor: I think the coding style for this project is to use braces around if blocks (aside from 1-liners)

File: package.json [hunk 2]
```diff
@@ -32,9 +32,8 @@
   },
   "devDependencies": {
     "lodash": "^2.4.1",
-    "jstransform": "^6.2.0",
+    "jsx-loader": "0.13.2",
```
Comment (jordangarcia) [inline package.json]: why change this?

File: src/store.js (hunk 1 above)
Comment (jordangarcia) [inline src/store.js]: @loganlinn please wrap with braces

```js
if (result === undefined) {
  throw new Error("Store handler must return a value, did you forget a return statement")
}
```

File: src/store.js [hunk 3]
```diff
@@ -68,7 +72,8 @@ class Store {
    * Binds an action type => handler
    */
   on(actionType, handler) {
-    this.__handlers = this.__handlers.set(actionType, handler)
+    console.assert(actionType, "Store.on called without actionType");
```
Comment (jordangarcia) [inline src/store.js]: not too comfortable with using `console.assert` here

I think it'd be find to throw an error.

I will probably add a `__DEV__` define in webpack in the near future and we can wrap these types of invariants in that flag that will compile out in prod bundles.

File: package.json (hunk 2 above)
Comment (appsforartists) [inline package.json]: It keeps pace with jstransform better.  (It's written by a Facebook alumnus and is commonly used for transpilation in the React community.)

It could also be worth looking into [babel](https://babeljs.io).  It's going to support more transformations.  Long term, I expect jstransform to be deprecated in favor of babel.

File: src/store.js (hunk 1 above)
Comment (appsforartists) [inline src/store.js]: :+1:

File: src/store.js (hunk 3 above)
Comment (appsforartists) [inline src/store.js]: Curious: what's bad about `console.assert`?

File: package.json (hunk 2 above)
Comment (jordangarcia) [inline package.json]: I'd prefer to keep this change in another PR.

I've been playing a little bit with `babel` and tend to like it.  Would want a little more usage before making the switch.

File: src/store.js (hunk 3 above)
Comment (jordangarcia) [inline src/store.js]: 1.  It doesn't seem like a standard https://developer.mozilla.org/en-US/docs/Web/API/console/assert
2.  I think the intended behavior here is to throw an error as this should be an invariant in the code.  Given that most people register stores at the initialization of your app throwing an error seems appropriate as you will always see it.

File: src/store.js (hunk 1 above)
Comment (appsforartists) [inline src/store.js]: Fixed.

File: package.json (hunk 2 above)
Comment (appsforartists) [inline package.json]: That was a bit lazy on my part.  Sorry about that.

File: src/store.js (hunk 3 above)
Comment (appsforartists) [inline src/store.js]: FWIW, the `console` API is supported by all the major browser engines.  Maybe it's not formally specified?

I'm used to using assertions to verify the state of the world, so I didn't even really consider whether an assertion or a conditional error was more appropriate when I wrote that.  I don't have a strong opinion either way, so it's now a conditional error.

Just now, I quickly tested the differences between the two.  It looks like errors thrown by `console.assert` break out of `try/catch`.  So, that's a good difference to know about…

File: src/store.js [hunk 4]
```diff
@@ -68,7 +73,11 @@ class Store {
    * Binds an action type => handler
    */
   on(actionType, handler) {
-    this.__handlers = this.__handlers.set(actionType, handler)
+    if (!actionType) {
+      throw new Error("Store.on called without a value for actionType.")
+    }
+
+    this.__handlers = this.__handlers.set(actionType.toString(), handler)
```
Comment (jordangarcia) [inline src/store.js]: hm what are the cases that we'd want to coerce this to a string?

Comment (appsforartists) [inline src/store.js]: After our chat yesterday, I've removed this.

**Comment by jordangarcia** [comment]
@appsforartists as mentioned in the issue I could not replicate this bug.  It seemed only to come up when you return `undefined` from `getInitialState()` which is exactly what the code is trying to prevent.

**Comment by appsforartists** [comment]
If I run into this again, I'll reopen.
